### PR TITLE
[JENKINS-32134]: Improved the Build Other Projects help message for p…

### PR DESCRIPTION
**Pull request by Dohbedoh**
Added a simple tip to specify downstream project located inside a folder.

@reviewbybees

### CODE REVIEW

File: war/src/main/webapp/help/project-config/downstream.html [hunk 1]
```diff
@@ -6,4 +6,8 @@
   Other than the obvious use case where you'd like to build other projects that have a dependency
   on the current project, this can also be useful to split a long build process in to multiple
   stages (such as the build portion and the test portion).
+
+  <p>
+    Projects located inside folders can be specified using an absolute path starting with "/" or a
```
Comment (oleg-nenashev) [inline war/src/main/webapp/help/project-config/downstream.html]: Folders Plugin is not a part of the Jenkins core. If you reference it, there should be a hyperlink to the plugin at least.

It also worths to reference other similar project types (e.g. Multibranch)

**Comment by oleg-nenashev** [comment]
:bug: Folders plugin is not referenced, it may confuse users

File: war/src/main/webapp/help/project-config/downstream.html (hunk 1 above)
Comment (Dohbedoh) [inline war/src/main/webapp/help/project-config/downstream.html]: I agree. I am gonna reference them.

**Comment by jglick** [comment]
:bee: though better would be to replace the textbox with a dropdown as requested in a longstanding JIRA RFE which I cannot easily look up at the moment.

File: war/src/main/webapp/help/project-config/downstream.html [hunk 2]
```diff
@@ -6,4 +6,11 @@
...
   on the current project, this can also be useful to split a long build process in to multiple
   stages (such as the build portion and the test portion).
+  <p>
+    When using
+    <a href="https://wiki.jenkins-ci.org/display/JENKINS/CloudBees+Folders+Plugin">CloudBees Folder Plugin</a>
+    or other plugins depending on it and organizing jobs into folders such as
+    <a href="https://wiki.jenkins-ci.org/display/JENKINS/Multi-Branch+Project+Plugin">Multi-Branch Project Plugin</a>,
+    projects located inside folders can be specified using an absolute path starting with "/" or a
+    relative path from the current project location.
 </div>
```
Comment (recena) [inline war/src/main/webapp/help/project-config/downstream.html]: :ant: Although `</p>` is not required, it is recommendable.

**Comment by KostyaSha** [comment]
Referencing all groupItem implementations sounds bad imho. Please define core terminology i.e. "GroupItem" and update plugins. This will be very objective.

File: war/src/main/webapp/help/project-config/downstream.html (hunk 2 above)
Comment (Dohbedoh) [inline war/src/main/webapp/help/project-config/downstream.html]: Good to know. I wasn't sure about this, I was following the format already in place.

**Comment by Dohbedoh** [comment]
I understand your point with the core terminology. As I will come up with something more objective. But don't you think we should at least mention the Folder plugin as an example?

File: war/src/main/webapp/help/project-config/downstream.html (hunk 2 above)
Comment (recena) [inline war/src/main/webapp/help/project-config/downstream.html]: *...following the format already in place.* it is not always a good option :smile:

**Comment by oleg-nenashev** [comment]
:ant: The text is too complex IMHO. It also focuses on Folders Plugin. My recommendation would be to describe the generic problem and then provide a list of examples.

File: war/src/main/webapp/help/project-config/downstream.html [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
-  on the current project, this can also be useful to split a long build process in to multiple
-  stages (such as the build portion and the test portion).
+    <p>
+        Other than the obvious use case where you'd like to build other projects that have a dependency
+        on the current project, this can also be useful to split a long build process in to multiple
+        stages (such as the build portion and the test portion).
+    </p>
+
+    <p>
+        A project located inside an ItemGroup need to be specified by using a path. It can be an absolute
```
Comment (daniel-beck) [inline war/src/main/webapp/help/project-config/downstream.html]: `ItemGroup` might be correct but not suitable for user documentation.

**Comment by daniel-beck** [comment]
Maybe *nested items* could be used in the description to refer to things in other things?

**Comment by aheritier** [comment]
@Dohbedoh can we move forward this PR ?

**Comment by aheritier** [comment]
thanks @olivergondza

**Pull request merged**